django/views/generic/edit.py

```diff
@@ -225,12 +225,26 @@ def get_success_url(self):
                 "No URL to redirect to. Provide a success_url.")
 
 
-class BaseDeleteView(DeletionMixin, BaseDetailView):
+class BaseDeleteView(DeletionMixin, FormMixin, BaseDetailView):
     """
     Base view for deleting an object.
 
     Using this base class requires subclassing to provide a response mixin.
     """
+    form_class = Form
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        form = self.get_form()
+        if form.is_valid():
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def form_valid(self, form):
+        success_url = self.get_success_url()
+        self.object.delete()
+        return HttpResponseRedirect(success_url)
```

Comment:
I wonder if it makes sense to add a hook (with better name) that could be called here and in `DeletionMixin`:
```python
class DeletionMixin:
    ...

    def _delete(self):
        if not getattr(self, 'object', None):
            self.object = self.get_object()
        success_url = self.get_success_url()
        self.object.delete()
        return HttpResponseRedirect(success_url)

    def delete(self, request, *args, **kwargs):
        return self._delete()

class BaseDeleteView(DeletionMixin, FormMixin, BaseDetailView):
    ...
    def form_valid(self, form):
        return self._delete()    
```

Reply:
Yeah... I could go either way here. At what point is it easier just to read it in place than go look for it in another class? (There's no definite answer there I think.)

Reply:
Agreed, the current form is more readable.

Reply:
OK, super. Thanks — I didn't change it 😄 

(My overriding feeling is that we have enough hooks already here, and than this one wouldn't ever be used anywhere...) 

But 👍 Thanks.